### ISSUE TITLE
Omnisharp process crashing in travis

### ISSUE THREAD

**Issue by akshita31** [OmniSharp/omnisharp-roslyn#1256]
The recent builds of omnisharp-vscode that consume the latest omnisharp from the master branch have been failing. On trying we realized that the versions till 1.32.2-beta.21 were working fine and the ones from
1.32.2-beta.33 onwards are failing. 

Here is the link to the failing build: https://travis-ci.org/OmniSharp/omnisharp-vscode/builds/410545826

**Comment by filipw** [comment]
beta.33 only has this change https://github.com/OmniSharp/omnisharp-roslyn/pull/1250 which is not related (not packaged into) OmniSharp distro. It was released from here https://travis-ci.org/OmniSharp/omnisharp-roslyn/jobs/407928780

So this shouldn't be related I think.

**Comment by filipw** [comment]
Oh actually, now I see. There is also version beta.31, which was never released for linux, because the release build failed at the time. 
That one is based on https://github.com/OmniSharp/omnisharp-roslyn/pull/1237 which is a much wider set of changes (albeit only updates).

Since beta.31 was never there for linux, beta.33 was the first time this particular changeset was released.

**Comment by filipw** [comment]
Here is what I see - when running on embedded mono

```
System.TypeLoadException: Could not load type of field 'McMaster.Extensions.CommandLineUtils.CommandLineApplication:_validationErrorHandler' (36) due to: Could not load file or assembly 'System.ComponentModel.DataAnnotations, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. assembly:System.ComponentModel.DataAnnotations, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 type:<unknown type> member:(null) signature:<none>
  at OmniSharp.Stdio.StdioCommandLineApplication..ctor () [0x00000] in <d3a78c5edca1483eaa0e46ffac11a90f>:0 
  at OmniSharp.Stdio.Driver.Program+<>c__DisplayClass0_0.<Main>b__0 () [0x00006] in <24c31f05c7ed4566b78400b5558caab0>:0 
  at OmniSharp.HostHelpers.Start (System.Func`1[TResult] action) [0x0001c] in <d571fd245c0a45a3925672c34247975f>:0 
```

When running on global mono (I have 5.12) everything is fine. This is definitely related to https://github.com/OmniSharp/omnisharp-roslyn/pull/1250 as this type `McMaster.Extensions.CommandLineUtils.CommandLineApplication` (and the whole package) was only introduced then, and looks like it needs dependencies we didn't need before.

@DustinCampbell do you know if `System.ComponentModel.Annotations` is being trimmed from the embedded mono we have?

**Comment by DustinCampbell** [comment]
No, we don't copy `System.ComponentModel.DataAnnotations` into the embedded Mono because nothing used it. Why did we need to upgrade CommandLineUtils to this forked version? Shouldn't the old one have worked fine?

**Comment by filipw** [comment]
> Why did we need to upgrade CommandLineUtils to this forked version? Shouldn't the old one have worked fine?

that is a good question 😃I think it was just a sweeping upgrade of all the packages involved (@david-driscoll?)

That said, you are right, we do not take advantage of any of the new extra features of this new package so I opened https://github.com/OmniSharp/omnisharp-roslyn/pull/1259 as a proposal to revert back to the old package and not have to modify the embedded Mono (and not have to grow its size too).

The only thing is that the old package `Microsoft.Extensions.CommandLineUtils` is no longer supported by MS - which is why the fork exists in the first place. But if we every have any bugs reported or issues with dealing with the command line args in OmniSharp, we can take have a look then. For now indeed there is no benefit.

**Comment by DustinCampbell** [comment]
Cool. That's probably the right thing in the short term. In the longer term, I'll take a look at moving the scripts that I use to create the embedded Mono into the OmniSharp repo. That's a good first step towards democratizing the embedded Mono so it isn't always gated on work from me.

**Comment by DustinCampbell** [comment]
We should probably add some tests that activate OmniSharp via command line args as well.

**Comment by DustinCampbell** [comment]
FWIW, there are tons of warnings on build with Mac/Linux since moving to net461.

```
/Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/xbuild/15.0/Microsoft.Common.targets/ImportAfter/Microsoft.Common.Mono.After.targets(37,9): warning MSB3912: INTERNAL WARNING: Could not find the replacement assembly (System.Globalization.Extensions.dll) for the Windows specific reference /Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/xbuild/Microsoft/Microsoft.NET.Build.Extensions/net461/lib/System.Globalization.Extensions.dll. Please file a bug report at https://bugzilla.xamarin.com . [/Users/dustincampbell/projects/omnisharp-roslyn/src/OmniSharp.Abstractions/OmniSharp.Abstractions.csproj]
```

**Comment by filipw** [comment]
Yep I saw them too. I tried with Mono 5.14 preview as well, but they are still there.

**Comment by marek-safar** [comment]
Should be fixed by https://github.com/mono/msbuild/commit/393268fe29c3f0b53911d82c13bd33b06e9023c5

**Comment by DustinCampbell** [comment]
Thanks @marek-safar !

**Comment by filipw** [comment]
This has been fixed by #1260
